modules/api/main.tf

```diff
@@ -1,5 +1,124 @@
 locals {
   app_name = "nerves_hub_api"
+  ecs_resources = var.alb ? [
+    aws_ecs_service.api_public_ecs_service[0].cluster,
+    "arn:aws:ecs:${var.region}:${var.account_id}:task-definition/nerves-hub-${terraform.workspace}-api:*",
+    ] : [
+    aws_ecs_service.api_ecs_service[0].cluster,
+    "arn:aws:ecs:${var.region}:${var.account_id}:task-definition/nerves-hub-${terraform.workspace}-api:*",
+  ]
+  ssm_prefix = "nerves_hub_api"
+
+  ecs_shared_env_vars = <<EOF
+    { "name" : "ENVIRONMENT", "value" : "${terraform.workspace}" },
+    { "name" : "APP_NAME", "value" : "${local.app_name}" },
+    { "name" : "HOST", "value" : "${aws_ssm_parameter.nerves_hub_api_ssm_host.value}" },
+    { "name" : "CLUSTER", "value" : "${aws_ssm_parameter.nerves_hub_api_ssm_cluster.value}" },
+    { "name" : "S3_BUCKET_NAME", "value" : "${aws_ssm_parameter.nerves_hub_api_ssm_s3_bucket_name.value}" }
```

Comment:
I dont know why bucket names are stored in SSM parameters, but sure that is a valid a way to do things, just adds an extra resource we don't need since we can pass these variables around in terraform.

But, one thing to be aware of with ECS task Definitions. 

They take [Environment vars](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ecs-taskdefinition-containerdefinitions.html#cfn-ecs-taskdefinition-containerdefinition-environment) (raw string values) and [secrets](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ecs-taskdefinition-containerdefinitions.html#cfn-ecs-taskdefinition-containerdefinition-secrets). The secrets are references to where to pull the secret from and will be evaluated when the container starts up. These static Environment values get resolved at terraform apply time. 

So this opens us up to an edge case where we apply terraform and everything is working fine. If we then manually use the AWS web console to update the ssm parameter that change won't make it to the task definition since it has a static value set. AND if we terraform apply again it will revert that ssm parameter value back to its original. (our other modules we have a lifecycle policy to ignore the actual value so that doesn't happen)

Reply:
Good call out. I'll look into that further